hpx/util/detail/pack_traversal_async_impl.hpp

```diff
@@ -275,14 +300,15 @@ namespace util {
         {
             Frame frame_;
             tuple<Hierarchy...> hierarchy_;
-            bool detached_;
+            std::atomic<bool>& detached_;
```

Comment:
* The usage of the reference is a great simplification 👍 
* From my point of understanding making the `detached_` variable atomic doesn't make any difference here since as described earlier the variable isn't shared across threads because we create a local variable for every resumption. So this can be ellided for sure. My assumption is that I forgot to propagate the value of the variable back somewhere. Probably the `atomic` can be converted back to a normal bool.

Reply:
Ok, I will verify whether converting to a plain `bool` is sufficient.